File: WalletTracker/app/src/main/java/com/chtima/wallettracker/adapters/TransactionAdapter.java
package com.chtima.wallettracker.adapters;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.chtima.wallettracker.R;
import com.chtima.wallettracker.models.Transaction;
import com.chtima.wallettracker.models.TransactionType;

import java.text.SimpleDateFormat;
import java.util.List;
import java.util.Locale;

public class TransactionAdapter extends RecyclerView.Adapter<TransactionAdapter.ViewHolder> {

    private final Context context;
    private final List<Transaction> list;

    public TransactionAdapter(Context context, List<Transaction> list) {
        this.context = context;
        this.list = list;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(context)
                .inflate(R.layout.transaction_card_line, parent, false));
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Transaction item = list.get(position);

        holder.title.setText(item.getDate());

        if(item.type == TransactionType.EXPENSE){
            holder.sum.setText("-" + item.sum);
            holder.sum.setTextColor(context.getResources().getColor(R.color.coral, null));
        }else {
            holder.sum.setText("+" + item.sum);
            holder.sum.setTextColor(context.getResources().getColor(R.color.lime, null));
        }
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    @SuppressLint("NotifyDataSetChanged")
    public void updateList(List<Transaction> list){
        this.list.clear();
        this.list.addAll(list);
        this.notifyDataSetChanged();
    }

    class ViewHolder extends RecyclerView.ViewHolder{

        final TextView title;
        final TextView sum;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            this.title = itemView.findViewById(R.id.title);
            this.sum = itemView.findViewById(R.id.sum);
        }
    }

}

File: WalletTracker/app/src/main/java/com/chtima/wallettracker/adapters/TransactionAdapter.kt
package com.chtima.wallettracker.adapters

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.chtima.wallettracker.R
import com.chtima.wallettracker.models.Transaction
import com.chtima.wallettracker.models.Transaction.TransactionType

class TransactionAdapter constructor (private val context: Context, private val list: MutableList<Transaction>) :
    RecyclerView.Adapter<TransactionAdapter.ViewHolder>() {


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(context).inflate(R.layout.transaction_card_line, parent, false))
    }

    override fun getItemCount(): Int {
        return list.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item: Transaction = list[position];

        holder.title.text = item.title
        if(item.type == TransactionType.EXPENSE){
            "-${item.sum}".also { holder.sum.text = it }
            holder.sum.setTextColor(context.getColor(R.color.coral));
        }else {
            "+${item.sum}".also { holder.sum.text = it }
            holder.sum.setTextColor(context.getColor(R.color.lime));
        }
    }

    @SuppressLint("NotifyDataSetChanged")
    fun updateList(newList: List<Transaction>){
        list.clear()
        list.addAll(newList)
        notifyDataSetChanged()
    }

    class ViewHolder(private val itemView: View) : RecyclerView.ViewHolder(itemView) {
        var title: TextView = itemView.findViewById(R.id.title);
        var sum: TextView = itemView.findViewById(R.id.sum);
    }

}
